### PR TITLE
Add unit tests for method security. terasolunaorg/terasoluna-tourreservation#213

### PR DESCRIPTION
**Pull request by kazuki43zoo**
I've added test cases for `@PostAuthorize`.
Please review terasolunaorg/terasoluna-tourreservation#213 .

### CODE REVIEW

File: terasoluna-tourreservation-domain/src/test/java/org/terasoluna/tourreservation/domain/service/reserve/ReserveServiceImplSecurityTest.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    ReserveRepository mockReserveRepository;
+
+    @Inject
+    ReserveService reserveService;
+
+    @Inject
+    JodaTimeDateFactory dateFactory;
+
+    @Before
+    @After
```
Comment (making) [inline terasoluna-tourreservation-domain/src/test/java/org/terasoluna/tourreservation/domain/service/reserve/ReserveServiceImplSecurityTest.java]: Why both `@Before` and `@After` are required?

Comment (kazuki43zoo) [inline terasoluna-tourreservation-domain/src/test/java/org/terasoluna/tourreservation/domain/service/reserve/ReserveServiceImplSecurityTest.java]: Added  `@Before` for safely.
However, `@Before` is not required. I will remove it.

**Comment by kazuki43zoo** [comment]
Reflect the review comment.
Please review again.

File: terasoluna-tourreservation-domain/src/test/java/org/terasoluna/tourreservation/domain/service/reserve/ReserveServiceImplSecurityTest.java (hunk 1 above)
Comment (making) [inline terasoluna-tourreservation-domain/src/test/java/org/terasoluna/tourreservation/domain/service/reserve/ReserveServiceImplSecurityTest.java]: Is `@Before` better?
I think all test using `SecurityContextHolder` should clear the context at the beginning of the test.

**Comment by kazuki43zoo** [comment]
I considered about setup/cleanup timing of mock data.

As result, i modified test case class as follows: 

* Set up a mock authentication user at the test case class initialization phase using `@BeforeClass`. 
  In other words, mock authentication is sharing at all test case methods in the test case class.
* Clean up a mock reservation data at the before of test case execution using `@Before`. And, set up a mock reservation data for test scenario at the test case method.
* Clean up a mock authentication user at the test case class destruction phase using `@AfterClass`.

Please review again.

**Comment by making** [comment]
please backport

**Pull request merged**